### PR TITLE
Document mergify and issue before PR

### PR DESCRIPTION
**Pull request by ravwojdyla**
Add mergify and "issue before PR" in contributing docs.

Re: #125 and #107

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -257,5 +257,12 @@ without warnings. You can do that locally with::
 Review process
 --------------
 
-Pull requests will be reviewed by a project maintainer. All changes to *sgkit* require
-approval by at least one maintainer.
+All pull requests require pre-existing issue. If you plan to submit a PR,
```
Comment (jeromekelleher) [inline docs/contributing.rst:260]:
```suggestion
All pull requests require a pre-existing issue. If you plan to submit a PR,
```

**Review by jeromekelleher — APPROVED** [review]
LGTM

File: docs/contributing.rst [hunk 2]
```diff
@@ -257,5 +257,12 @@ without warnings. You can do that locally with::
...
+please create an issue first to describe and discuss your idea/plan/problem.
+
+Pull requests will be reviewed by a project maintainer. All changes to *sgkit*
+require approval by at least one maintainer.
+
+We use `mergify <https://mergify.io/>`_ to automate PR flow, should you like
```
Comment (jeromekelleher) [inline docs/contributing.rst:266]: Actually, come to think of it, I don't think this is the right protocol. The person who opens the PR shouldn't decide whether something should be automerged, it should be an admin who makes this decision. So, if one admin reviews the PR and thinks it's OK, they mark it automerge. If they think other admins should have a look, they don't.

Comment (jeromekelleher) [reply docs/contributing.rst:266]: I think it'll really confuse more casual contributors too to have to understand whether they should mark as auto-merge or not.

Comment (ravwojdyla) [reply docs/contributing.rst:266]: Sure - that sound fine to me as well.

Comment (ravwojdyla) [reply docs/contributing.rst]: @jeromekelleher updated.

File: docs/contributing.rst (hunk 1 above)
Comment (ravwojdyla) [reply docs/contributing.rst]: Fixed - thanks!

**Review by jeromekelleher — APPROVED** [review]
Thanks @ravwojdyla, LGTM!

**Comment by ravwojdyla** [comment]
FYI, and it looks like we still need to configure the mergify installation on `pystatgen` to include this repo.

File: docs/contributing.rst [hunk 3]
```diff
@@ -257,5 +257,12 @@ without warnings. You can do that locally with::
 Review process
 --------------
 
-Pull requests will be reviewed by a project maintainer. All changes to *sgkit* require
-approval by at least one maintainer.
+All pull requests require a pre-existing issue. If you plan to submit a PR,
+please create an issue first to describe and discuss your idea/plan/problem.
```
Comment (tomwhite) [inline docs/contributing.rst:261]: Maybe add the bit about mentioning the issue in the PR comments here from https://github.com/pystatgen/sgkit/issues/125

**Review by tomwhite — APPROVED** [review]

File: docs/contributing.rst (hunk 3 above)
Comment (ravwojdyla) [reply docs/contributing.rst]: Done

**Comment by ravwojdyla** [comment]
Ok - so that worked fine.

**Comment by jeromekelleher** [comment]
Ooh, why did mergify merge master in here? Wouldn't it be better to rebase onto to the base branch? I think we've got things the wrong way around - we should rebase to bring the branch up to date and merge to merge it in. That way the network graph is much more comprehensible:

Screenshot from 2020-08-20 17-33-28


vs

Screenshot from 2020-08-20 17-35-11

Maybe I should make this an issue.

**Comment by ravwojdyla** [comment]
@jeromekelleher actually, doing it this way is:
 * similar to doing a rebase merge manually (before mergify was enabled) (which was we have agreed to do in #26)
 * using `strict_method: rebase` which what you are referring to, and what `tskit` does is actually not recommended by mergify itself: https://doc.mergify.io/actions.html#strict-rebase

**Pull request merged**